pkg/driver/controller.go

```diff
@@ -68,20 +70,56 @@ type controllerService struct {
 	volumeLocks   *util.VolumeLocks
 }
 
+// Provider holds information from the cloud provider.
+type Provider struct {
+	// PowerVSCloudInstanceID is IBM Power VS service instance id
+	PowerVSCloudInstanceID string `gcfg:"powerVSCloudInstanceID"`
+	// PowerVSZone is IBM Power VS service zone
+	PowerVSZone string `gcfg:"powerVSZone"`
+}
+
+// CloudConfig is the ibm cloud provider config data.
+type CloudConfig struct {
+	// [provider] section
+	Prov Provider `gcfg:"provider"`
+}
+
 var (
 	NewPowerVSCloudFunc = cloud.NewPowerVSCloud
 )
 
 // newControllerService creates a new controller service
 // it panics if failed to create the service
 func newControllerService(driverOptions *Options) controllerService {
-	klog.V(4).Infof("retrieving node info from metadata service")
-	metadata, err := cloud.NewMetadataService(cloud.DefaultKubernetesAPIClient)
-	if err != nil {
-		panic(err)
+	var (
+		cloudInstanceId string
+		zone            string
+	)
+	if driverOptions.cloudconfig != "" {
+		var cloudConfig CloudConfig
+		config, err := os.Open(driverOptions.cloudconfig)
+		if nil != err {
+			panic(err)
+		}
+		defer config.Close()
+
+		err = gcfg.FatalOnly(gcfg.ReadInto(&cloudConfig, config))
+		if nil != err {
+			panic(err)
+		}
+		cloudInstanceId = cloudConfig.Prov.PowerVSCloudInstanceID
+		zone = cloudConfig.Prov.PowerVSZone
```

Comment:
There is possibility that these values can be a empty string if user doesn't pass the values in the config file, lets add an additional check to ensure we have the fields set(non-empty)

Reply:
Fixed.